### ISSUE TITLE
Return type nullability lint check reports bogus errors

### ISSUE THREAD

**Issue by crdev** [google/closure-compiler#656]
For pure virtual methods, like

```
Foo.prototype = {
    /**
     * @return {?Object}
     */
    method: function()
    {
        throw new Error("Not implemented.");
    }
}
```

the following error is reported:

```
ERROR - The return type of the function "method" is nullable, but it always returns a non-null value. Consider making the return type non-nullable
```

Also, a similar error is reported for methods that override nullable superclass methods but always return non-null:

```
/**
 * @constructor
 * @extends {Foo}
 */
function Bar() {}
Bar.prototype = {
    /**
     * @return {?Object}
     */
    method: function()
    {
        return {};
    }
}
```

Perhaps, these types of methods should not be affected by this check.
